### PR TITLE
Core: Improve webpack chunk names

### PR DESCRIPTION
**Pull request by yannbf**
Issue: N/A

**UPDATE: Removed the prefetching and kept the nice chunk names. See rationale and planned followup below**

## What I did

With the new story store and stories being lazy loaded, the initial loading experience is great but the experience when switching stories is degraded. This PR adds two things:

**1 - Webpack chunk names.** 
Important to identify what's actually being downloaded rather than random names like:

and have more meaningful names like:

**2 - Prefetched chunks**

Here's a video comparison of the user experience without and with prefetching enabled in a Slow 3G setting.

**Before (without prefetching):**

https://user-images.githubusercontent.com/1671563/139343097-54c73143-aea9-4763-9b52-57c3e32bb4f0.mov

**After (with prefetching):**

https://user-images.githubusercontent.com/1671563/139343105-fdc6acdf-246d-42a9-be35-f2b2460f6088.mov

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [ ] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by tmeasday** [comment]
> And that should also not impact the initial user experience. The scripts will be fetched and cached for later use only at idle time and at low priority for the browser. 

@yannbf did you test it out?

I just tried it in Chromatic's Storybook and I am seeing a definitely slowdown in the time to render the first story (from around 1 -> 2.5s in my rough example) . I'm not sure why the browser isn't waiting to fetch the files but it seems to launch of a massive batch of requests straight away.

I would be leaning towards maybe not including it given that, unless we can find some way to make the browser deprioritize it.

**Comment by shilman** [comment]
we'll follow up in 6.5 with a custom approach that eagerly loads all links that are visible AFTER the entire storybook has loaded, to avoid the slowdown mentioned above.

thanks for the awesome POC @yannbf !!! 💯

**Comment by shilman** [comment]
Closing this with #16546 as a followup

**Comment by tmeasday** [comment]
@shilman I think we may as well keep the nicer bundle name part of this?

File: lib/core-common/src/utils/to-importFn.ts [hunk 1]
```diff
@@ -32,6 +32,8 @@ export function toImportFnPart(specifier: NormalizedStoriesSpecifier) {
 
         const pathRemainder = path.substring(${directory.length + 1});
         return import(
+          /* webpackPrefetch: true */
```
Comment (shilman) [inline lib/core-common/src/utils/to-importFn.ts:35]:
```suggestion
```

**Comment by shilman** [comment]
merging the modified PR without prefetching

**Pull request merged**